core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -549,6 +630,32 @@ public boolean advanceExact(int target) throws IOException {
         }
     }
 
+    /** wraps a {@link Rectangle} to include an optional field name for queries */
+    public static class GeoBBox extends Rectangle {
+        /** optional fieldname for queries */
+        private String fieldName;
```

Comment:
Why do we need the fieldname as part of the parsed bbox structure. As far as I can see this gets only parsed when an inner `field` parameter is seen, the use of which seems both undocumented and it also doesn't get rendered when using the GeoBoundingBoxQueryBuilder#doXContent method. I have a hope that we can remove this maybe?